### ISSUE TITLE
New check: SingleSpace

### ISSUE THREAD

**Issue by mkordas** [sevntu-checkstyle/sevntu.checkstyle#334]
Multiple spaces can be used only for indentation and inside string literals. In all other contexts this check should raise a violation when more than one subsequent space character is used.

Examples:
```java
/** Sample    Javadoc */ //violation
class      Sample { //violation
                   int count; //long indentation - OK
                   int             number; //violation
                   String text = "             "; // OK
}
```
More real samples can be found in checkstyle/checkstyle#728

**Comment by romani** [comment]
that Check should validate comments and javadoc too, most likely it should not be a TreeWalker listener, as there is not need to be aware of java structure.

**Comment by mkordas** [comment]
@romani, are you sure that this check does not need to know Java structure?

I can think of quite complicated cases where regexp probably won't help:
```java
        String s1 = "\"    +    "; // no violation, all inside one string
        String s2 = "\\"   +   ""; // violation, two strings
```

**Comment by romani** [comment]
I never put a preference to RegExp for parsing.
Lets do that with java parsing if we are not sure - that will be a bit slower bit BUT more reliable. Users need reliable work from Check, speed is not a first priority.

**Comment by robertwhitebit** [comment]
I'll try it!
Any classes/checks I should take a look at which helps implementing or are similar to this check? :eyeglasses:

**Comment by romani** [comment]
I do see any similar Check, looks like you do not need visitToken, as you need all tokens, so you travetse tree yourself from method beginTree

**Comment by romani** [comment]
Please move all discussions to sepatate topic https://groups.google.com/forum/m/#!forum/sevntu-checkstyle , to avoid making mess in issue

**Comment by romani** [comment]
https://github.com/checkstyle/checkstyle/issues/1007 that check should somehow consider such code or not ...... we need to think about that.

**Comment by wstomv** [comment]
Adding options for enforcing single spacing to each of the various whitespace checks individually would make checking on amount of whitespace better tunable. Also see issue #2090 and my post in https://groups.google.com/forum/m/#!forum/sevntu-checkstyle

**Comment by romani** [comment]
merged at https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/384
